emulator/blockchain.go

```diff
@@ -1182,6 +1208,34 @@ func (b *Blockchain) executeBlock() ([]*types.TransactionResult, error) {
 		}
 	}
 
+	script := templates.ReplaceAddresses(
+		systemChunkTransactionTemplate,
+		templates.Environment{
+			RandomBeaconHistoryAddress: b.GetChain().ServiceAddress().Hex(),
+		},
+	)
+	serviceKey := b.ServiceKey()
+	serviceAccount, _ := b.getAccount(flowgo.Address(b.serviceKey.Address))
+	tx := flowsdk.NewTransaction().
+		SetScript([]byte(script)).
+		SetGasLimit(9999).
+		AddAuthorizer(serviceKey.Address).
+		SetProposalKey(serviceKey.Address, 1, serviceAccount.Keys[1].SeqNumber).
+		SetPayer(serviceKey.Address).
+		SetReferenceBlockID(flowsdk.Identifier(b.pendingBlock.parentID))
+
+	signer, err := serviceKey.Signer()
+	if err != nil {
+		return nil, err
+	}
+
+	err = tx.SignEnvelope(serviceKey.Address, 1, signer)
+	if err != nil {
+		return nil, err
+	}
+	txn := convert.SDKTransactionToFlow(*tx)
```

Comment:
I am not sure about this part, @janezpodhostnik knows better, maybe we have another way to execute this transaction without modifying SeqNumber of the service account, this may break some tests maybe? if they were assuming seqNumber increasing if they send transaction by one

Reply:
@bluesign  I tried to use the predefined service key, as the proposal key, but that wouldn't work. Even a transaction for account creation is signed by the service account, which means that in the same block, we can't have two transactions from the service account, with the same proposal key. That's why I added a completely new account key. This in fact did break some tests, but I've fixed them. Not sure if there's an alternative for this, because I would always get a conflict on the sequence number, when using the same proposal key.

Reply:
I think one option is to send proposer at `0x0` (actually not setting proposer at all is same effect I guess) and temporarily disable transaction validation. I think somehow real network is not setting proposer and signature for this transaction too

Reply:
Aha, I see. I wasn't able to find how this transaction body is constructed in the `flow-go` repository.

Reply:
I tried to make a PR to your repo but didn't allow me @m-Peter 

Basically what I did was : 

- move chunkTransaction execution to `commitBlock` ( as we can skip blocks with commitBlock without executing)
- rollback test changes with results ( length of results ) 
- execute chunkTransaction with:
```go
ctx := fvm.NewContextFromParent(
		b.vmCtx,
		fvm.WithAuthorizationChecksEnabled(false),
		fvm.WithSequenceNumberCheckAndIncrementEnabled(false),
		fvm.WithRandomSourceHistoryCallAllowed(true),
	)
```

Reply:
That's nice :clap: 
So the checks are enabled/disabled on the context itself. I thought it was on transaction body :sweat_smile: 
Not sure why you can't open a PR, it is just a fork of this one.
Let me see if I can apply your suggestions :100:

Reply:
I am not sure about events part: `b.pendingBlock.events = append(b.pendingBlock.events, output.Events...)`
not sure how mainnet manage systemchunkevents, @janezpodhostnik can chime in there.

Reply:
I do not think that the `RandomBeaconHistory` contract emits any events. Maybe I misunderstood something though?

Reply:
oh I remember there was a discussion on events there, if there is none there is no concern.

Reply:
Hats off to @bluesign for his suggestion :100: 
That clearly made the changes more contained, as there are less files affected.
Take a look and let me know if this is what you had in mind.

Reply:
great work @m-Peter